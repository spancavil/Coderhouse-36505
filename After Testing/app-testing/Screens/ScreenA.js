import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import { useState } from 'react';
import React from 'react'

const ScreenA = ({navigation}) => {

    const [status, setStatus] = useState(false);

    return (
        <View style={styles.container}>
            <Text>Coderhouse - testing with React Nativo</Text>
            <TouchableOpacity
                testID='myButton'
                onPress={() => setStatus(!status)}>
                    <Text>Change status</Text>
            </TouchableOpacity>
            <Text testID='myText'>{status.toString()}</Text>
            <TouchableOpacity
                testID='navButton'
                onPress={() => navigation.navigate("Screen B")}>
                    <Text>Go to Screen B</Text>
            </TouchableOpacity>
        </View>
    )
}

export default ScreenA

const styles = StyleSheet.create({})