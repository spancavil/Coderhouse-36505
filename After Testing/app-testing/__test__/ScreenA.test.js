import React from 'react';
import renderer from 'react-test-renderer';
import { act } from 'react-test-renderer';

import ScreenA from '../Screens/ScreenA';
let component

//Mockeamos la navigation
const navigation = {
    navigate: jest.fn()
}

describe('<Screen A/>', () => {
    beforeEach(()=> {
        component = renderer.create(<ScreenA navigation={navigation}/>);
    })
    
    it('The component is defined', () => {
        const tree = component
        expect(tree).toBeDefined();
    });

    it('Renders correctly (by snapshot)', () => {
        const tree = component
          .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('The main view has 4 childs', () => {
        const tree = component.toJSON();
        expect(tree.children.length).toBe(4);
    });

    it ("Button pressed and changed status", ()=> {
        const button = component.root.findByProps({testID: "myButton"}).props;
        //Simulamos que se presionÃ³ el button. Utilizamos act porque se triggerea un re-render
        act(()=> button.onPress())
        const text = component.root.findByProps({testID: "myText"}).props;
        expect(text.children).toEqual('true')
    })

    it("Testing de navigation", ()=> {
        const button = component.root.findByProps({testID: "navButton"}).props;
        //No usamos act porque no triggerea un rerender.
        button.onPress();
        expect(navigation.navigate).toBeCalledWith('Screen B')
    })
});