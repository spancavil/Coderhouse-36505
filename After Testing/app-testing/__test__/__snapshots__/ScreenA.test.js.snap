// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen A/> Renders correctly (by snapshot) 1`] = `
<View>
  <Text>
    Coderhouse - testing with React Nativo
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="myButton"
  >
    <Text>
      Change status
    </Text>
  </View>
  <Text
    testID="myText"
  >
    false
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="navButton"
  >
    <Text>
      Go to Screen B
    </Text>
  </View>
</View>
`;
